packages/headless/src/features/tab-set/tab-set-selectors.ts

```diff
@@ -16,3 +16,17 @@ export const selectActiveTab = createSelector(
     return '';
   }
 );
+
+export const selectActiveTabExpression = createSelector(
+  (state?: TabSetState) => state,
+  (tabSetState?: TabSetState): string => {
+    if (!tabSetState) {
+      return '';
+    }
+    if (!selectActiveTab(tabSetState)) {
+      return '';
+    }
+
+    return tabSetState[selectActiveTab(tabSetState)].expression;
+  }
+);
```

Comment:
You're calling the `selectActiveTab` twice here, and it already checks for empty state so I don't think you have to double check?

```suggestion
export const selectActiveTabExpression = createSelector(
  (state?: TabSetState) => state,
  (tabSetState?: TabSetState): string => {
    const activeTabId = selectActiveTab(tabSetState);

    return activeTabId ? tabSetState[activeTabId].expression : '';
  }
);
```

Reply:
@ThibautNazare This one is the most important comment I believe.